### PR TITLE
Optimized all resource tracker to stop tracking services that are not re...

### PR DESCRIPTION
**Pull request by kolkoo**
...sources.Fixed a bug with HttpTracker removedService method not being called. Fixed a bug with application not being marked dirty when removing a resource.

Optimized Configuration,ServletConfiguration and ApplicationConfiguration updates to not recreate the JerseyContext but instead update the existing objects. Reworked/extended tests.

Signed-off-by: IVAN GEORGIEV ILIEV <ivan.iliev@musala.com>

### CODE REVIEW

**Comment by hstaudacher** [comment]
Thanks will review in a few days. We are currently in a release phase which does not allow me to spend time for this right now ;(

**Comment by kolkoo** [comment]
That's ok for now we are using a locally built version of the publisher. When this makes it to a release we'll adapt :)

File: bundles/com.eclipsesource.jaxrs.publisher/src/com/eclipsesource/jaxrs/publisher/internal/Activator.java [hunk 1]
```diff
@@ -1,14 +1,10 @@
...
+ * Copyright (c) 2012,2015 EclipseSource and others. All rights reserved. This program and the
+ * accompanying materials are made available under the terms of the Eclipse Public License v1.0
  * which accompanies this distribution, and is available at
- * http://www.eclipse.org/legal/epl-v10.html
- *
- * Contributors:
- *    Holger Staudacher - initial API and implementation
- *    ProSyst Software GmbH. - compatibility with OSGi specification 4.2 APIs
- *    Ivan Iliev - added ServletConfigurationTracker
+ * http://www.eclipse.org/legal/epl-v10.html Contributors: Holger Staudacher - initial API and
```
Comment (kaikreuzer) [inline bundles/com.eclipsesource.jaxrs.publisher/src/com/eclipsesource/jaxrs/publisher/internal/Activator.java]: @kolkoo You seem to have messed up the formatting of the license headers...

**Comment by hstaudacher** [comment]
Don't worry. I will fix them during review ;)

**Comment by hstaudacher** [comment]
Sorry, could not manage to integrate it so far. I'm also on vacation fro 2 weeks now. Will do it right after I return. Sorry for the delay

**Comment by hstaudacher** [comment]
working on it today

**Comment by kolkoo** [comment]
ok let me know how it goes :)

**Comment by hstaudacher** [comment]
Merged with minor changes. Can you verify that the current master version works for you?

**Comment by kolkoo** [comment]
Hi, yes I will check it out by the end of this week.

**Comment by kolkoo** [comment]
Sorry for the delayed response, everything is looking good for us. I am now using a built version from the latest master - and have added some slf4j logging for debugging purposes to it and all is looking good. Maybe we can consider adding some logging to the connector or what is your opinion about that?

**Comment by hstaudacher** [comment]
Nope, no logging :). If needed users can use the Jersey tracing feature, see https://jersey.java.net/documentation/latest/user-guide.html#appendix-properties

**Comment by kolkoo** [comment]
Yes, I thought so :P

**Comment by kaikreuzer** [comment]
@hstaudacher Any plans to do a new release which includes these optimizations?

**Comment by hstaudacher** [comment]
Yes, I jus tneed to upgrade to Jersey 2.18. Probably next week I will find some time

**Pull request closed without merging**